build/Dockerfile

```diff
@@ -1,46 +1,289 @@
-		/etc/nginx/stream-conf.d \
-	&& apt-get update \
+RUN apt-get update \
 	&& apt-get install -y libcap2-bin \
-	&& setcap 'cap_net_bind_service=+ep' /usr/sbin/nginx \
-	          'cap_net_bind_service=+ep' /usr/sbin/nginx-debug \
-	&& chown -R nginx:0 /etc/nginx \
-			    /var/cache/nginx \
-			    /var/lib/nginx \
-	&& apt-get remove --purge -y libcap2-bin \
-	&& rm /etc/nginx/conf.d/* \
 	&& rm -rf /var/lib/apt/lists/*
 
-COPY internal/configs/version1/nginx.ingress.tmpl \
-	internal/configs/version1/nginx.tmpl \
-	internal/configs/version2/nginx.virtualserver.tmpl \
-	internal/configs/version2/nginx.transportserver.tmpl /
+
+############################################# Base image for Alpine #############################################
+FROM nginx:$NGINX_VERSION-alpine AS alpine
+ARG IC_VERSION
+
+LABEL version=$NGINX_VERSION-alpine-$IC_VERSION
+LABEL maintainer="NGINX Docker Maintainers <docker-maint@nginx.com>"
+
+RUN apk add --no-cache libcap \
+	&& rm -rf /var/cache/apk/*
+
+
+############################################# Base image for Debian with NGINX Plus #############################################
+FROM debian:buster-slim AS debian-plus
+ARG IC_VERSION
+ENV NGINX_PLUS_VERSION 23-1~buster
+ENV NGINX_NJS_VERSION 23+0.5.0-1~buster
+
+LABEL version=debian-nginx-plus-$NGINX_PLUS_VERSION-$IC_VERSION
+LABEL maintainer="NGINX Docker Maintainers <docker-maint@nginx.com>"
+
+RUN set -x \
+	&& apt-get update \
+	&& apt-get install --no-install-recommends --no-install-suggests -y ca-certificates gnupg wget \
+	&& wget https://nginx.org/keys/nginx_signing.key \
+	&& gpg --no-default-keyring --keyring nginx_keyring.gpg --import nginx_signing.key \
+	&& gpg --no-default-keyring --keyring nginx_keyring.gpg --export > /etc/apt/trusted.gpg.d/nginx_signing.gpg \
+	&& echo "Acquire::https::plus-pkgs.nginx.com::Verify-Peer \"true\";" >> /etc/apt/apt.conf.d/90nginx \
+	&& echo "Acquire::https::plus-pkgs.nginx.com::Verify-Host \"true\";" >> /etc/apt/apt.conf.d/90nginx \
+	&& echo "Acquire::https::plus-pkgs.nginx.com::SslCert     \"/etc/ssl/nginx/nginx-repo.crt\";" >> /etc/apt/apt.conf.d/90nginx \
+	&& echo "Acquire::https::plus-pkgs.nginx.com::SslKey      \"/etc/ssl/nginx/nginx-repo.key\";" >> /etc/apt/apt.conf.d/90nginx \
+	&& echo "Acquire::https::plus-pkgs.nginx.com::User-Agent  \"k8s-ic-$IC_VERSION-apt\";" >> /etc/apt/apt.conf.d/90nginx \
+	&& printf "deb https://plus-pkgs.nginx.com/debian buster nginx-plus\n" > /etc/apt/sources.list.d/nginx-plus.list \
+	&& apt-get purge --auto-remove -y gnupg wget \
+	&& rm -rf /var/lib/apt/lists/*
+
+RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	apt-get update && apt-get install -y apt-transport-https libcap2-bin nginx-plus=${NGINX_PLUS_VERSION} nginx-plus-module-njs=${NGINX_NJS_VERSION} \
+	&& apt-get purge --auto-remove -y apt-transport-https \
+	&& rm -rf /var/lib/apt/lists/*
```

Comment:
what about removing files `/etc/apt/apt.conf.d/90nginx /etc/apt/sources.list.d/nginx-plus.list` like we do here  https://github.com/nginxinc/kubernetes-ingress/blob/master/build/DockerfileForPlus#L46 ?

Reply:
they're needed in the ap stage